File: Best Time to Buy and Sell Stock/Best Time to Buy and Sell Stock.java
public class Solution {
    public int maxProfit(int[] prices) {
        int max = 0;
        int tmp = 0;
        for (int i = 0; i < prices.length - 1; ++i) {
            tmp += prices[i + 1] - prices[i];
            max = Math.max(max, tmp);
            if (tmp < 0) tmp = 0;
        }
        return max;
    }
}

File: Best Time to Buy and Sell Stock/Best Time to Buy and Sell Stock.cpp
class Solution {
public:
    int maxProfit(vector<int> &prices) {
        int res = 0;
        int tmp = 0;
        for (int i = 0; i < (int)prices.size() - 1; ++i) {
            tmp += prices[i + 1] - prices[i];
            res = max(res, tmp);
            if (tmp < 0) tmp = 0;
        }
        return res;
    }
};